shared/models/videos/video-resolution.enum.ts

```diff
@@ -1,7 +1,59 @@
+import { VIDEO_TRANSCODING_FPS } from '../../../server/initializers'
```

Comment:
You must not import constants in the models.

I would suggest to create a model representing `VIDEO_TRANSCODING_FPS`, in the constants use `const VIDEO_TRANSCODING_FPS: MyModel = { ... }` and use it `function getTargetBitrate (resolution: VideoResolution, fps: number, transcodingFPS: MyModel)`

Reply:
I'm not sure how you want this to be implemented exactly. Right now I have this model, should I just return that from `ffmpeg-utils.getVideoFileFps()`? That means FPS would always be exactly 10/30/60, but nothing in between.
```
export enum VideoTranscodingFps {
    FPS_60 = 60,
    FPS_30 = 30,
    FPS_10 = 10
  }
```

Reply:
I see, then I replace `if (fps == 60)` with `if (fps == fpsTranscodingConstants.MAX)`. 

There's one problem related to this: if a video has 60 fps, it will be converted with 4400 Kbps, but if it has 59 fps, it only gets 3300 Kbps. But maybe this is something to be fixed later, this PR is already huge.

Reply:
:+1: 

You will have to use KEEP_ORIGIN_FPS_RESOLUTION_MIN too :)

Reply:
Hmm I'm not sure what bitrate should be used for 480p 60 fps video etc. How about we leave this for the future fps rework? I'd like to finish this PR at some point :P

Reply:
Fixed :)